### ISSUE TITLE
Allow explicitly setting a page's navbar label (rather than using the page title)

### ISSUE THREAD

**Issue by kristinashu** [MozillaFoundation/foundation.mozilla.org#8000]
### Describe the feature request
via marcwalsh in Slack
> [marc] Is it possible to change the text in the nav bar for a specific page. I’m creating a page for MozFest House: Amsterdam and hoping to set the page title as ‘MozFest House: Amsterdam’ but keep the nav bar text as ‘House’.

> [kristina] just to clarify - you're requesting that the page title in a tab is the "Title tag" and not the "Slug"?
![](https://user-images.githubusercontent.com/1068783/145471498-83ca266b-132c-4dd4-82ca-6694dd736a88.png)

> [marc] I think a field for itself it would be great. I'd imagine the slug would be lowercase, the title tag would be longer, but the nav in this instance would be short.
> Another example would be us having Tickets in the nav, but maybe something longer in the title/header of the page.

### Additional context
Since this update would affect the entire website, design & devs should look into what the best practices and decide from there.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/TP1-256)

**Comment by danielfmiranda** [comment]
Hi @kristinashu @anilkanji!

After taking a look at some best practices online, such as this article that Anil had sent me: https://www.nngroup.com/articles/navigation-you-are-here/, it does appear that having a different name for the page title vs the navigation goes against best practices.

For example, here is a screenshot that the article above uses to show "good use of location indicators"


<img width="593" alt="Screen Shot 2022-02-23 at 2 31 36 PM" src="https://user-images.githubusercontent.com/18314510/155420472-943b7210-4089-45b3-9e15-c5c615931781.png">

At the left side of the page, they use a navigation panel to show all the subsections or child pages of the current page, with the titles matching in figure 2 (bottom) and 4.


A closer example would be this screenshot from the BBC site, also from the same article, as figure 2 in this screenshot shows they are using a navigation panel quite similar to ours:

<img width="662" alt="image" src="https://user-images.githubusercontent.com/18314510/155421773-b3b1f36b-c18b-473c-b4ab-c4c68620a4c3.png">

At first glance it appears that the highlighted navigation choice and page title are different, however figure 3 shows that they are displaying the title of the selected tab in the navbar on the page as well, though *slightly* different and less prominent then the heading of the article.

While they are different, I think the difference between "Tech" and "Technology" is quite minimal and easily recognizable. And while this can be said for "House" and "Mozfest House: Amsterdam" from this request, I am concerned that if we add this option for users, it opens the possibility of having very different titles, making navigation hard to follow. 



However I did want to mention that after taking a look through the code, I see that this is something we can add quite easily if we choose to move forward. 

It would be a matter of adding a new field, and saying "If we have an alternate nav title set, render that. If not, render the title".


Thank you for taking a look at my comment and I look forward to discussing this 😄

**Comment by kristinashu** [comment]
Thank you so much @danielfmiranda for looking into this! It does make sense that it should be consistent and giving staff the tools to make the nav name and the page title *anything* could become confusing and problematic - both for staff and our users.

@danielfmiranda I think this ticket is good to close. @marcwalsh please reach out if you have any concerns.

**Comment by danielfmiranda** [comment]
Closing due to discussion above, please reach out if there are any concerns. Thank you!

**Comment by mtdenton** [comment]
Reopening this at the request of @marcwalsh

We shouldn't allow the possibility of nav text to deviate very far from the page title for UX and accessibility reasons, but what if we validated the contents of the nav text against the page title to make sure that at least one word from the page title is present in the nav text and nothing else is allowed? ex:

Page Title: "MozFest House Kenya"

Acceptable nav text: "House", "Kenya", "House Kenya", "Mozfest Kenya", "Kenya Mozfest"
Unacceptable nav text: "Tickets", "Kenya Tickets", " Mozfest Nairobi"

Seems like this would keep everything on the up and up. @danielfmiranda, let's discuss this when it's convenient.

**Comment by kristinashu** [comment]
That would be amazing!

**Comment by jhonatan-lopes** [comment]
IMO, I would prefer to have a separate page type (say, a `Mozfest House page`) for this as a child of Mozfest Homepage where the nav is fixed as **Home**, since this has been the case twice. Implementing a fuzzy match between the title and a potential nav text sounds like a can of worms :worm:

**Comment by kristinashu** [comment]
Thank you for adding your input @jhonatan-lopes!

Also, for future reference, I'll add the context from Marc's request in slack today:

> I'm wondering if I could open this ticket back open for discussion as Audrey brought this up to me earlier. We're facing the same issue again where we're looking to send people to the 'House' page as a landing page to purchase a ticket to MozFest House Kenya. The page is called House so that it fits well in the nav bar, but the heading on the page should be MozFest House Kenya. It's quite limiting to have to find a balance between a page title - it would be great if the 'Header' field in the template changed the heading but not what appears in the nav, Example below:

> ![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/e9640675-321c-42cf-8313-3597eae1d3b8)
![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/fb3ddfe3-5a76-433a-8188-1c48821d2b42)

**Comment by kristinashu** [comment]
Note to self: Users can't see the main nav selection on mobile e.g https://foundation.mozilla.org/en/who-we-are/

<img width="447" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/d9f18f9d-a75b-467b-bfb1-c10c62cc54cc">

**Comment by kristinashu** [comment]
Looks the the NNg site seems to do something like what Tyler mentioned. The nav uses a single word "Articles" but then the heading is longer and more descriptive "Articles & Videos" (not that the NNg site is the most beautiful and engaging of sites lol). 

https://www.nngroup.com/articles/
<img width="1225" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/363bbfc5-0f2d-4e11-bee5-ffb2a3f07249">
<img width="1449" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/1cb883e6-15ef-4295-a49f-0aab9a3a7b08">

**Comment by kristinashu** [comment]
Design talked about this during crit and here are a few notes:

Currently, the Heading field on banner pages doesn't do anything
<img width="1345" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/57bdb268-91d2-4220-8e23-90905621c9a1">

If we make it so the Heading field replaces the page title field then
- We need to add clear help text to the page so staff understand what each of the fields does
- Clear guidelines on how to write proper headings
- Could we use workflows to approve of the headings? (the auto-validation to see if the nav word is in the heading might not always make for the best headings)
- Could we make the Heading field collapsed by default so people will only override it if they really want to?
- Could we start by only implementing this on the mozfest site?

Note that Heading (and the title) appear on Basic pages:
<img width="1344" alt="image" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/1068783/561f5593-9d7c-4135-8e6d-af46c595e749">

**Comment by simont-cr** [comment]
@kristinashu Are the things discussed in this ticket part of the new navigation effort, making this obsolete, or do we need to keep this ticket as a reference of what we're trying to achieve once the nav is done?

**Comment by kristinashu** [comment]
Right now we're focused on the main nav. We hope to look at the sub nav later this year so keeping this info would be helpful. I'll update it so it's just on design (not eng) tho.

**Comment by data-sync-user** [comment]
➤ Simon Acosta Torres commented:

Daniel Miranda I will abandon this ticket for the time being as I don’t think this will be relevant to keep on the backlog. Feel free to restore if you disagree. Thanks!
